kover-maven-plugin/src/main/kotlin/kotlinx/kover/maven/plugin/mojo/KoverInstrumentMojo.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.maven.plugin.mojo
+
+import kotlinx.kover.maven.plugin.Constants
+import kotlinx.kover.maven.plugin.mojo.abstracts.AbstractKoverMojo
+import org.apache.maven.artifact.Artifact
+import org.apache.maven.plugin.MojoExecutionException
+import org.apache.maven.plugins.annotations.LifecyclePhase
+import org.apache.maven.plugins.annotations.Mojo
+import org.apache.maven.plugins.annotations.Parameter
+import java.io.File
+
+/**
+ * Add JVM agent argument to the JVM in which tests are running.
+ */
+@Mojo(name = "instrumentation", defaultPhase = LifecyclePhase.INITIALIZE, threadSafe = true)
+class KoverInstrumentMojo : AbstractKoverMojo() {
+    @Parameter(property = "plugin.artifactMap", required = true, readonly = true)
+    private lateinit var pluginArtifactMap: Map<String, Artifact>
+
+    /**
+     * Property name to pass java agent argument to the JVM in which tests are running.
+     *
+     * `argLine` by default.
+     */
+    @Parameter(property = "kover.agentPropertyName", defaultValue = Constants.AGENT_ARG_PARAMETER)
+    private lateinit var agentPropertyName: String
+
+    @Parameter
+    var uninstrumentedClasses: List<String>? = null
+
+    @Parameter(defaultValue = "\${project.build.directory}/${Constants.BIN_REPORT_PATH}")
+    lateinit var binaryReportFile: File
+
+    override fun doExecute() {
+        val propertyName = agentPropertyName ?: Constants.AGENT_ARG_PARAMETER
```

Comment:
Elvis does not have effect on `lateinit` vars. Access to uninitialized var will throw a `kotlin.UninitializedPropertyAccessException`.
I see you handle this differently in your code, e.g. `IcReportMojo.icFile` is not checked at all
or `AbstractCoverageTaskMojo.aggregate` is not a `lateinit var` at all

Reply:
My bad, it's a dead code